modules/1plusXRtdProvider.js

```diff
@@ -0,0 +1,211 @@
+    bidder =>
+      SUPPORTED_BIDDERS.includes(bidder) && adUnitBidders.includes(bidder)
+  );
+  if (!bidders.length) {
+    throw new Error('No supported bidder found in config parameter bidders');
+  }
+
+  return { customerId, timeout, bidders };
+}
+
+/**
+ * Gets the URL of Profile Api from which targeting data will be fetched
+ * @param {Object} config
+ * @param {string} config.customerId
+ * @returns {string} URL to access 1plusX Profile API
+ */
+const getPapiUrl = ({ customerId }) => {
+  // https://[yourClientId].profiles.tagger.opecloud.com/[VERSION]/targeting?url=
+  const currentUrl = encodeURIComponent(window.location.href);
+  const papiUrl = `https://${customerId}.profiles.tagger.opecloud.com/${PAPI_VERSION}/targeting?url=${currentUrl}`;
+  return papiUrl;
+}
+
+/**
+ * Fetches targeting data. It contains the audience segments & the contextual topics
+ * @param {string} papiUrl URL of profile API
+ * @returns {Promise} Promise object resolving with data fetched from Profile API
+ */
+const getTargetingDataFromPapi = (papiUrl) => {
+  return new Promise((resolve, reject) => {
+    const requestOptions = {
+      customHeaders: {
+        'Accept': 'application/json'
+      }
+    }
+    const callbacks = {
+      success(responseText, response) {
+        resolve(JSON.parse(response.response));
+      },
+      error(error) {
+        reject(error);
+      }
+    };
+    ajax(papiUrl, callbacks, null, requestOptions)
+  })
+}
+
+/**
+ * Prepares the update for the ORTB2 object
+ * @param {Object} targetingData Targeting data fetched from Profile API
+ * @param {string[]} segments Represents the audience segments of the user
+ * @param {string[]} topics Represents the topics of the page
+ * @returns {Object} Object describing the updates to make on bidder configs
+ */
+export const buildOrtb2Updates = ({ segments = [], topics = [] }) => {
+  const userData = {
+    name: '1plusX.com',
+    segment: segments.map((segmentId) => ({ id: segmentId }))
+  };
+  const site = {
+    keywords: topics.join(',')
+  };
+  return { userData, site };
+}
+
+/**
+ * Merges the targeting data with the existing config for bidder and updates
+ * @param {string} bidder Bidder for which to set config
+ * @param {Object} ortb2Updates Updates to be applied to bidder config
+ * @param {Object} bidderConfigs All current bidder configs
+ * @returns {Object} Updated bidder config
+ */
+export const updateBidderConfig = (bidder, ortb2Updates, bidderConfigs) => {
+  if (!SUPPORTED_BIDDERS.includes(bidder)) {
+    return null;
+  }
+  const { site, userData } = ortb2Updates;
+  const bidderConfigCopy = mergeDeep({}, bidderConfigs[bidder]);
+
+  const currentSite = deepAccess(bidderConfigCopy, 'ortb2.site')
+  const updatedSite = mergeDeep(currentSite, site);
+
+  const currentUserData = deepAccess(bidderConfigCopy, 'ortb2.user.data') || [];
+  const updatedUserData = [
+    ...currentUserData.filter(({ name }) => name != userData.name),
+    userData
+  ];
```

Comment:
In `ortb2.user.data`; replacing the element bearing `1plusX.com` name with the element defined at `buildOrtb2Updates::const userData`

Reply:
it is confusing you say you only support AppNexus and Magnite, yet you seem to be writing to an accessible object for all bidders here.